stdlib/public/runtime/Casting.cpp

```diff
@@ -2755,6 +2755,56 @@ swift::swift_conformsToProtocol(const Metadata *type,
   goto recur;
 }
 
+/// Return the type metadata for a given type name.
+///
+/// This is a temporary implementation to support some Foundation use cases;
+/// it will only lookup a type that directly implements a protocol as only
+/// those types will have conformance table entries.
+static const Metadata *
+_metadataForName(const llvm::StringRef typeName) {
+  auto &C = Conformances.get();
+  const Metadata *foundMetadata = nullptr;
+
+  pthread_mutex_lock(&C.SectionsToScanLock);
+
+  unsigned sectionIdx = 0;
+  unsigned endSectionIdx = C.SectionsToScan.size();
+
+  // TODO: implement typeName->metadata cache
+  // TODO: support looking up types for which there is no conformance entry
+  //       and generic/non-nominal types
+  for (; sectionIdx < endSectionIdx; ++sectionIdx) {
+    auto &section = C.SectionsToScan[sectionIdx];
+    for (const auto &record : section) {
+      if (auto metadata = record.getCanonicalTypeMetadata()) {
+        auto N = nameForMetadata(metadata, true);
+        if (N == typeName) {
+          foundMetadata = metadata;
+          break;
+        }
+      }
+    }
+  }
+
+  pthread_mutex_unlock(&C.SectionsToScanLock);
+
+#if SWIFT_OBJC_INTEROP
+  if (foundMetadata == nullptr) {
+    foundMetadata =
+      static_cast<const Metadata *>(objc_lookUpClass(typeName.c_str());
```

Comment:
Accidental commit of this, haven't tested on OS X yet. Is this desired functionality?